thingsboard_gateway/gateway/tb_gateway_service.py

```diff
@@ -146,6 +147,7 @@ def __init__(self, config_file=None):
         global log
         log = logging.getLogger('service')
         log.info("Gateway starting...")
+        self.__duplicate_detector = DuplicateDetector(self)
```

Comment:
I think that there is no reason to pass a link to the whole Gateway instance (in DuplicateDetector using only self.__available_connectors)

Reply:
ok

Reply:
Well, my way is encapsulated and strongly ensured that duplicate detector get the latest updated list of connector after possible remote updates and some other features what may be added in the future. Agree?

Reply:
Yes, but self.__available_connectors is a dict that is a mutable object, so I think it must work too.

Reply:
Done. Please, review again whole PR